Ghidra/Processors/WebAssembly/data/languages/WebAssembly.sinc

```diff
@@ -0,0 +1,426 @@
+}
+
+reftype: "reftype="^bytev is bytev {
+	export *[const]:1 bytev;
+}
+
+# An explicit no-op operation, to be used before
+# callother instructions to work around Ghidra issue #3390
+# (which causes incorrect decompilation if the first pcodeop
+# in a branch target instruction is a callother)
+macro nop() {
+	LR = LR;
+}
+
+macro trap() {
+<trap_loop>
+	halt_trap();
+	goto <trap_loop>;
+}
+
+macro doBranch(br_target) {
+	nop();
+	popCallOther(&t0);
+	pushCallOther(&t0);
+	goto br_target;
+}
+
+################################################################
+
+## Locals directives at the start of the function
+:.locals is count; ctx_is_directive=1; ctx_directive_type=$(DIRECTIVE_LOCALS) {
+}
+
+:.local "count="^count "type="^typeidx is count; typeidx; ctx_is_directive=1; ctx_directive_type=$(DIRECTIVE_LOCAL) {
+}
+
+## Control instructions
+:unreachable is opc=0x00; ctx_is_directive=0 {
+	trap();
+}
+
+:nop is opc=0x01; ctx_is_directive=0 {
+}
+
+:block "blocktype="^ULeb128 is opc=0x02; ULeb128; ctx_is_directive=0 {
+}
+:loop "blocktype="^ULeb128 is opc=0x03; ULeb128; ctx_is_directive=0 {
+}
+:if "blocktype="^ULeb128 is opc=0x04; ULeb128; br_target; sp1_32; ctx_is_directive=0 {
+	if(sp1_32 != 0) goto inst_next;
+	doBranch(br_target);
+}
+:else is opc=0x05; br_target; ctx_is_directive=0 {
+	doBranch(br_target);
+}
+
+:end is opc=0x0b; ctx_is_return=1; ctx_is_directive=0 {
+	nop();
+	popCallOther(&o0);
+	return [LR];
+}
+:end is opc=0x0b; ctx_is_return=0; ctx_is_directive=0 {
+}
+
+:br br_target is opc=0x0c; ULeb128; br_target; ctx_is_directive=0 {
+	doBranch(br_target);
+}
+:br_if br_target is opc=0x0d; ULeb128; br_target; sp1_32; ctx_is_directive=0 {
+	if(sp1_32 == 0) goto inst_next;
+	doBranch(br_target);
+}
+
+:.case ctx_case_index br_target is ULeb128; br_target; ctx_case_index; ctx_is_directive=1; ctx_directive_type=$(DIRECTIVE_CASE) {
+	if(BRTableInput != ctx_case_index) goto inst_next;
+	doBranch(br_target);
+}
+
+:.default br_target is ULeb128; br_target; ctx_is_directive=1; ctx_directive_type=$(DIRECTIVE_DEFAULT) {
+	doBranch(br_target);
+}
+
+:br_table is opc=0x0e; ULeb128; sp1_32; ctx_is_directive=0 {
+	BRTableInput = sp1_32;
+}
+
+:return is opc=0x0f; ctx_is_directive=0 {
+	nop();
+	popCallOther(&o0);
+	return [LR];
+}
+:call br_target is opc=0x10; ULeb128; br_target; ctx_is_directive=0 {
+	nop();
+	popCallOther(&i0);
+	call br_target;
+	pushCallOther(&o0);
+}
+:call_indirect "type="^typeidx tableref is opc=0x11; typeidx; tableref; sp1_32; ctx_is_directive=0 {
+	nop();
+	addr:8 = &tableref + sext(sp1_32) * 8;
+	val:8 = *[table] addr;
+	popCallOther(&i0);
+	call [val];
+	pushCallOther(&o0);
+}
+
+## Reference Instructions
+:ref.null reftype is opc=0xd0; reftype; sp0_64; ctx_is_directive=0 {
+	sp0_64 = 0;
+}
+:ref.is_null is opc=0xd1; sp1_32; sp1_64; ctx_is_directive=0 {
+	res:4 = zext(sp1_64 == 0);
+	sp1_32 = res;
+}
+:ref.func funcidx is opc=0xd2; funcidx; br_target; sp0_64; ctx_is_directive=0 {
+	sp0_64 = &br_target;
```

Comment:
at first glance this looks weird, is this just `ctx_br_target`

Reply:
Good catch - actually, I should have used `ref.func br_target` rather than `ref.func funcidx` in order to create an operand reference to the function. I'll make that change.